src/bindgen/cdecl.rs

```diff
@@ -314,6 +318,19 @@ impl CDecl {
                             arg_ty.write(out, arg_ident, config);
                         }
                     }
+
+
+                    match layout {
+                        Layout::Vertical => write_vertical(out, config, args),
+                        Layout::Horizontal => write_horizontal(out, config, args),
+                        Layout::Auto => {
+                           if out.line_length_for_align() + out.measure(|out| write_horizontal(out, config,args)) > config.line_length {
```

Comment:
I'm not a fan of having to do the work twice in the common case, but we should be able to improve it. Looks good, thanks!

Reply:
what do you mean" do the work twice" ?